### PR TITLE
provider: Support default tags (resources aws_i*)

### PR DESCRIPTION
**Pull request by bflad**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Reference: https://github.com/hashicorp/terraform-provider-aws/issues/7926

Output from acceptance testing: Will be handled via CI, review, and daily acceptance testing

### CODE REVIEW

**Review by anGie44 — APPROVED** [review]
(merge conflict to address..otherwise LGTM 👍 )

File: aws/resource_aws_iam_policy.go [hunk 1]
```diff
@@ -162,10 +168,17 @@ func resourceAwsIamPolicyRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("path", policy.Path)
 	d.Set("policy_id", policy.PolicyId)
 
-	if err := d.Set("tags", keyvaluetags.IamKeyValueTags(policy.Tags).IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+	tags := keyvaluetags.IamKeyValueTags(policyRes.Tags).IgnoreAws().IgnoreConfig(ignoreTagsConfig)
```
Comment (bflad) [inline aws/resource_aws_iam_policy.go:171]:
```suggestion
	tags := keyvaluetags.IamKeyValueTags(policy.Tags).IgnoreAws().IgnoreConfig(ignoreTagsConfig)
```

File: aws/resource_aws_iam_instance_profile.go [hunk 2]
```diff
@@ -69,13 +69,18 @@ func resourceAwsIamInstanceProfile() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
-			"tags": tagsSchema(),
+			"tags":     tagsSchema(),
+			"tags_all": tagsSchemaComputed(),
```
Comment (anGie44) [inline aws/resource_aws_iam_instance_profile.go:73]: Ooo i missed on first review @bflad: just need to update this resource's Read method to set these attributes, hidden in the `instanceProfileReadResult` func

File: aws/resource_aws_imagebuilder_image.go [hunk 3]
```diff
@@ -134,17 +134,22 @@ func resourceAwsImageBuilderImage() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
-			"tags": tagsSchema(),
+			"tags":     tagsSchema(),
+			"tags_all": tagsSchemaComputed(),
```
Comment (anGie44) [inline aws/resource_aws_imagebuilder_image.go:138]: missed on first review @bflad: just need to update this resource's Read method to set these attributes

File: aws/resource_aws_imagebuilder_image_pipeline.go [hunk 4]
```diff
@@ -127,13 +127,18 @@ func resourceAwsImageBuilderImagePipeline() *schema.Resource {
 				Default:      imagebuilder.PipelineStatusEnabled,
 				ValidateFunc: validation.StringInSlice(imagebuilder.PipelineStatus_Values(), false),
 			},
-			"tags": tagsSchema(),
+			"tags":     tagsSchema(),
+			"tags_all": tagsSchemaComputed(),
```
Comment (anGie44) [inline aws/resource_aws_imagebuilder_image_pipeline.go:131]: missed on first review @bflad: just need to update this resource's Read method to set these attributes

File: aws/resource_aws_inspector_assessment_template.go [hunk 5]
```diff
@@ -71,8 +76,8 @@ func resourceAwsInspectorAssessmentTemplateCreate(d *schema.ResourceData, meta i
 
 	d.SetId(aws.StringValue(resp.AssessmentTemplateArn))
 
-	if v := d.Get("tags").(map[string]interface{}); len(v) > 0 {
-		if err := keyvaluetags.InspectorUpdateTags(conn, d.Id(), nil, v); err != nil {
+	if len(tags) > 0 {
+		if err := keyvaluetags.InspectorUpdateTags(conn, d.Id(), nil, tags); err != nil {
```
Comment (anGie44) [inline aws/resource_aws_inspector_assessment_template.go:80]: i _think_ in this case `tags` needs to be passed in as a map or else when `New` is called on it (when inside `InspectorUpdateTags`) , it'll be a new `KeyValueTags` with the values lost

Comment (bflad) [reply aws/resource_aws_inspector_assessment_template.go:80]: Submitted https://github.com/hashicorp/terraform-provider-aws/pull/19011 to cover this nonintuitive situation 👍

File: aws/resource_aws_iam_instance_profile.go (hunk 2 above)
Comment (bflad) [reply aws/resource_aws_iam_instance_profile.go:73]: Updated in https://github.com/hashicorp/terraform-provider-aws/pull/18723/commits/7dc456f1093f7a695dbbb67fa8dcfff3f9712875

File: aws/resource_aws_imagebuilder_image.go (hunk 3 above)
Comment (bflad) [reply aws/resource_aws_imagebuilder_image.go:138]: Updated in https://github.com/hashicorp/terraform-provider-aws/pull/18723/commits/7dc456f1093f7a695dbbb67fa8dcfff3f9712875

File: aws/resource_aws_imagebuilder_image_pipeline.go (hunk 4 above)
Comment (bflad) [reply aws/resource_aws_imagebuilder_image_pipeline.go:131]: Updated in https://github.com/hashicorp/terraform-provider-aws/pull/18723/commits/7dc456f1093f7a695dbbb67fa8dcfff3f9712875

**Pull request merged**